internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1600,8 +1600,8 @@ func (r *BMACReconciler) formatMCSCertificateIgnition(mcsCert string) (string, e
 
 func (r *BMACReconciler) validateWorkerForDay2(log logrus.FieldLogger, agent *aiv1beta1.Agent) bool {
 	// Only worker role is supported for day2 operation
-	if agent.Status.Role != models.HostRoleWorker || agent.Spec.ClusterDeploymentName == nil {
-		log.Debugf("Skipping spoke BareMetalHost reconcile for agent, role %s and clusterDeployment %s.", agent.Status.Role, agent.Spec.ClusterDeploymentName)
+	if agent.Spec.ClusterDeploymentName == nil {
```

Comment:
please remove the comment above this change 
"// Only worker role is supported for day2 operation" - it's not relevant anymore

Reply:
This PR is just my suggestion on what should change, it's not intended to be merged, someone else should write a proper PR and test this